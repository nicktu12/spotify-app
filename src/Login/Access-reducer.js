export function accessToken(state='', action){
  switch (action.type) {
  case 'ACCESS_TOKENS':
    return action.accessToken;
  default:
    return state;
  }
}

export function topArtists(state=[], action){
  switch (action.type) {
  case 'TOP_ARTISTS':
    return action.topArtists;
  default:
    return state;
  }
}

export function userInfo(state={}, action){
  switch (action.type) {
  case 'USER_INFO':
    return action.userInfo;
  default:
    return state;
  }
}

export function recentlyPlayed(state={}, action){
  switch (action.type) {
  case 'RECENTLY_PLAYED':
    return action.recentlyPlayed;
  default:
    return state;
  }
}
