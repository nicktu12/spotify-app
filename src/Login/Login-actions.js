export const testClickAction  = (string) => {
  return {
    type: 'TEST_CLICK',
    data: string
  }
}

export const apiStuffAction = (stuff) => {
  return {
    type: 'SAGAS_TEST_SUCCESS',
    stuff
  }
}


