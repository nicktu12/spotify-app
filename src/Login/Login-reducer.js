export function testClickAction(state = {}, action){
  switch (action.type) {
	  case 'TEST_CLICK':
		  return action.data
	  default:
		  return state;
  }
}

export function apiStuffAction(state={}, action){
  switch(action.type) {
    case 'SAGAS_TEST_SUCCESS':
	    return action.stuff
    default: 
	   return state;
  }
}
