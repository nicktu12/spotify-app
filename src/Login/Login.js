import React from 'react';
import { spotifyClientId } from '../Utilities/apiKey';

export const Login = () => (
  <div className='login-div'>
    <h2>Welcome to Statify!</h2>
    <div>
      <p>
        To see your Spotify listening statistics<br/> 
        please sign in to Spotify below
      </p>
      <p>
        <a href={
          `https://accounts.spotify.com/authorize/` + 
          `?client_id=${spotifyClientId}` +
          `&response_type=code&` + 
          `redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F` +
          `&scope=user-read-private%20user-read-email%20user-top-read` +
          `&state=34fFs29kd09`
        }> 
          <button>Login</button>
        </a>
      </p>
    </div>
  </div>
);
