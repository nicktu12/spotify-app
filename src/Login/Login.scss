@import '../Styles/Variables.scss';

.login-div {
	height: calc(100vh - 90px);
	box-sizing: border-box;
	border-radius: 0px 0px $border-rad $border-rad;
}

.login-div h2 {
  @include linear-grad($dark-bg-o, $dark-bg);
  color: $white;
  text-align: left;
  padding-left: 10px;
  padding-top: 50px;
	height: 91px;
	box-sizing: border-box;
	font-family: $header-font;
	font-size: 32px;
	text-align: center;
}

.login-div button {
  background-color: $spotify-green;
	border: none;
	padding: 2px 30px;
  color: $white;
  font-size: 16px;
	font-family: $header-font;
}

.login-div div {
  background-color: $dark-bg;
  color: $light-3;
  text-align: left;
  padding-left: 35px;
	overflow: scroll;
	height: calc(99vh - 174px);
	border-radius: 0px 0px $border-rad $border-rad;
  line-height: 32px;
	font-family: $body-font;
}

.login-div p {
	padding-top: 4%;
	font-family: $body-font;
	text-align: center;
}

