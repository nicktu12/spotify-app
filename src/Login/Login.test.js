import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';

describe('Login component tests', ()=>{
  it('should match snapshot render', ()=>{
    const wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  })
})
