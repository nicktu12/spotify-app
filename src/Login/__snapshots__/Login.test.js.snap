// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should match snapshot render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Welcome to Statify!
        </h2>,
        <div>
          <p>
            To see your Spotify listening statistics
            <br />
            please sign in to Spotify below
          </p>
          <p>
            <a
              href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
            >
              <button>
                Login
              </button>
            </a>
          </p>
        </div>,
      ],
      "className": "login-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Statify!",
        },
        "ref": null,
        "rendered": "Welcome to Statify!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              To see your Spotify listening statistics
              <br />
              please sign in to Spotify below
            </p>,
            <p>
              <a
                href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
              >
                <button>
                  Login
                </button>
              </a>
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "To see your Spotify listening statistics",
                <br />,
                "please sign in to Spotify below",
              ],
            },
            "ref": null,
            "rendered": Array [
              "To see your Spotify listening statistics",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "please sign in to Spotify below",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
              >
                <button>
                  Login
                </button>
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button>
                  Login
                </button>,
                "href": "https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              "type": "a",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Welcome to Statify!
          </h2>,
          <div>
            <p>
              To see your Spotify listening statistics
              <br />
              please sign in to Spotify below
            </p>
            <p>
              <a
                href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
              >
                <button>
                  Login
                </button>
              </a>
            </p>
          </div>,
        ],
        "className": "login-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Statify!",
          },
          "ref": null,
          "rendered": "Welcome to Statify!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                To see your Spotify listening statistics
                <br />
                please sign in to Spotify below
              </p>,
              <p>
                <a
                  href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
                >
                  <button>
                    Login
                  </button>
                </a>
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "To see your Spotify listening statistics",
                  <br />,
                  "please sign in to Spotify below",
                ],
              },
              "ref": null,
              "rendered": Array [
                "To see your Spotify listening statistics",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "please sign in to Spotify below",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09"
                >
                  <button>
                    Login
                  </button>
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button>
                    Login
                  </button>,
                  "href": "https://accounts.spotify.com/authorize/?client_id=6f67e11fa50a413f9bf17697789322aa&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=user-read-private%20user-read-email%20user-top-read%20playlist-modify-public%20playlist-modify-private&state=34fFs29kd09",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
                "type": "a",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
